jupyter/nmm-artificial-intelligence.ipynb

```diff
@@ -0,0 +1,167 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Künstliche Intelligenz\n",
+    "\n",
+    "Diese Studienarbeit implementiert zwei verschiedene Algorithmen als künstliche Intelligenz: Minimax und α-β-Pruning. Damit diese Algorithmen leichter wiederverwendet und mit verschiedenen Einstellungen ausgeführt werden können, wird eine abstrakte Superklasse angelegt. Diese bescheibt welche Funktionen nötig sind, damit eine Implementierung den Ansprüchen einer künstlichen Intelligenz für Mühle entspricht."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os.path\n",
+    "css = \"\"\n",
+    "if os.path.isfile(\"style.html\"):\n",
+    "    from IPython.core.display import HTML\n",
+    "    with open(\"style.html\", \"r\") as file:\n",
+    "        css = file.read()\n",
+    "HTML(css)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## BestMoves"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Die Klasse `BestMoves` beschreibt das Ergebnis, welches eine künstliche Intelligenz für Mühle erzeugen soll. Diese Klasse beschreibt den Rückgabewert der später definierten Funktion `bestMoves`. Sie hat drei Attribute:\n",
+    "\n",
+    "* `states` $\\subset States$;\n",
+    "* `value` $\\in \\mathopen[-1.0,1.0\\mathclose]$;\n",
+    "* `meta` ist ein `dict`, welches weitere Informationen wie beispielsweise die erreichte Rekursionstiefe oder die besuchten Zustände beinhalten kann."
```

Comment:
Ich finde `meta` nicht wirklich selbstsprechend. Vielleicht sowas wie `payload` oder `content`?

Reply:
Hmm, ich finde beide Alternativen auch nicht wirklich gut. Durch deine vorgeschlagene Wahl, klingt es so als sei das wichtige bei dem Ergebnis das `dict` mit Debug-Informationen. Aber wirklich interessant sind ja eigentlich die States und die Value.

Reply:
Stimmt schon, aber meta kann ja auch quasi alles bedeuten

Reply:
mit `metaData` könnte ich mich vielleicht noch anfreunden.
Was sagt denn @nclskfm dazu?

Reply:
> Debug-Informationen

`DebugInformation`?

Reply:
Habe es äquivalent zu den anderen in lowerCamelCase geschrieben